### PR TITLE
Fix missing client side validations for Challenge edit form

### PR DESCRIPTION
**Pull request by piotrekpasciak**
I'll add Image_file validations in separate PR.

### CODE REVIEW

File: app/views/challenges/fields/_challenge_rules_fields.html.erb [hunk 1]
```diff
@@ -24,3 +24,8 @@
     <%= f.text_area :additional_checkbox_text_markdown, class: "form-control txt-sml" %>
   <% end %>
 </div>
+
+<script>
+  // I have to refresh client_side_validations after adding the association
+  $('#challenges-form').enableClientSideValidations();
```
Comment (skbly7) [inline app/views/challenges/fields/_challenge_rules_fields.html.erb]: Is it possible to add it at a single place so we don't forget for new tabs in future? Maybe in challenge tab animation js code?

Comment (piotrekpasciak) [reply app/views/challenges/fields/_challenge_rules_fields.html.erb]: This has to be in this file, as if I move it elsewhere new nested forms . won't handle the validation.

Comment (piotrekpasciak) [reply app/views/challenges/fields/_challenge_rules_fields.html.erb]: https://github.com/AIcrowd/AIcrowd/blob/master/app/views/challenges/tabs/_rules_form.html.erb#L12

This link won't work correctly if I move it outside this partial.

**Comment by nemaniarjun** [comment]
You can check the PR for challenge rounds. I did something similar there
using a cocoon callback, maybe you can use that too.

**Comment by piotrekpasciak** [comment]
@nemaniarjun 
I think approach you're proposing won't work because this javascript line needs to be executed after association partial gets inserted to html.

If you want to proof I'm wrong go ahead and show me . working solution.

File: app/views/challenges/fields/_challenge_rules_fields.html.erb (hunk 1 above)
Comment (skbly7) [reply app/views/challenges/fields/_challenge_rules_fields.html.erb]: Callback addition somewhere like here with 1000 delay: https://github.com/AIcrowd/AIcrowd/blob/45f339b1af0a69f9576c94596a55dc3418887508/app/assets/javascripts/controllers/challenges_controller.js#L38

Comment (piotrekpasciak) [reply app/views/challenges/fields/_challenge_rules_fields.html.erb]: Screenshot 2020-01-28 at 18 31 56

@skbly7 Did you mean something like that?

Comment (piotrekpasciak) [reply app/views/challenges/fields/_challenge_rules_fields.html.erb]: @skbly7 It doesn't work for me. Are you sure this should work? Can we jump . on a call where I'll show you the edge case?

Comment (piotrekpasciak) [reply app/views/challenges/fields/_challenge_rules_fields.html.erb]: @skbly7  Showed me how to make it work properly.

Comment (piotrekpasciak) [reply app/views/challenges/fields/_challenge_rules_fields.html.erb]: I've updated the code to include your remark.

Comment (skbly7) [reply app/views/challenges/fields/_challenge_rules_fields.html.erb]: It looks great now!

**Pull request merged**